### DISCUSSION TITLE
Inventory Client - ACCESS DENIED

### DISCUSSION THREAD

**Discussion by sdaeric19** [saleweaver/python-amazon-sp-api#40] [General]
Hey guys, 
I'm struggling with the Inventory Client. 

My goal is to pull current inventory of all my ASINs. 

My tokens are fine as they work with order, sales, product clients

Here's my code
```
from sp_api.api import Orders, Reports, Sales, ProductFees, Products, Catalog, Inventories
from sp_api.base import SellingApiException, Marketplaces, Granularity
from datetime import datetime, timedelta, timezone
from sp_api.base.InventoryEnums import InventoryGranularity


# Inventory API
try:
    res = Inventories(marketplace=Marketplaces.GB).get_inventory_summary_marketplace(granularityType=InventoryGranularity.MARKETPLACE,
                                                                                     granularityId="A1F83G8C2ARO7P",
                                                                                     marketplaceIds="A1F83G8C2ARO7P")
    print(res.payload)  # json data
except SellingApiException as ex:
    print(ex)

```

However I'm getting access denied error on this and Catalog Client. Please can you help me?
```
[{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]

```

**Comment by saleweaver** [comment]
The token has to work with FBA, the other endpoints are unrelated to this. Can you check if the box for FBA is ticked in sellercentral.

**Reply by sdaeric19** [reply]
Is there another box to tick?

These are the roles I allocated to my app 
<img width="761" alt="Screenshot 2021-02-13 at 20 25 08" src="https://user-images.githubusercontent.com/78454351/107860832-a7a91780-6e39-11eb-8c5d-06af74b23aef.png">

**Reply by saleweaver** [reply]
Looks good, I haven't seen the box since some time. There's currently no converter from enum to string, you need to pass 
`InventoryGranularity.MARKETPLACE.value` or simply leave it out, since it is the only possible value for granularityType atm. Below request does the same as yours but should work:

```
.get_inventory_summary_marketplace() 

```

**Reply by sdaeric19** [reply]
It didn't work, I created a new app in Developer Central with new tokens, ticked all boxes and now everything stopped working (same error)  :(

Does it take some time to activate a new app?

**Reply by sdaeric19** [reply]
Also is it a problem if I have two apps in the Developer Central - both in draft stage but they both have the same ticked boxes?

**Reply by saleweaver** [reply]
Usually it doesn't matter how many apps you have. It is quite strange that this one doesn't work for you while other endpoints do; I don't have an immediate solution for this. I'll have a look tomorrow though.

**Reply by smilekudo** [reply]
Is there any updates @saleweaver ? i have same issue.

**Reply by sdaeric19** [reply]
@smilekudo  I haven't looked into it as I deemed inventory lower priority for now. I'll look into it later. But I'm looking into fees for now. 

Maybe @saleweaver  has an idea

**Reply by Zakir-Ayub** [reply]
@sdaeric19 Hi did you find a solution here, I am going through the same issue now

**Reply by saleweaver** [reply]
> @sdaeric19 Hi did you find a solution here, I am going through the same issue now

If that's the inventory summaries endpoint you're having problems with, and you're trying to get them for a Marketplace other than US, it won't work at the moment because Amazon doesn't support this yet for non US marketplaces.  
I've heard that this was to change, but that was some months back already...

**Reply by Zakir-Ayub** [reply]
oh, okay, makes sense then, I am trying for DE.

**Reply by kleczekr** [reply]
Hey @saleweaver were there any developments on this front? I'm having the same problem as others, with the error saying that I am unauthorized to access the requested resource. As an example, this request:

`res = Inventories(credentials = credentials, marketplace=Marketplaces.US).get_inventory_summary_marketplace(marketplaceIds=[Marketplaces.US.marketplace_id])`

Returns this error:

`[{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]`

The Orders endpoint on the other hand works perfectly fine with the same credentials. It shouldn't be an issue of the app permissions, since I ticked off the Inventory box (as well as all the other boxes for that matter).

The stores I'm handling operate only in the US, so region also shouldn't be a problem.

**Reply by kleczekr** [reply]
Just a quick update here: I managed an ad-hoc solution for my purpose using the reports API, which works.

When using the reports API, I've noticed that I had to request specifically FBA inventory reports rather than regular inventory reports. Might it be related to my issue regarding the Inventory API queries?

Thanks!
